### PR TITLE
TCP2 recv win and mss update 

### LINKED ISSUE #30367: TCP2 does not send our MSS to peer
**Is your enhancement proposal related to a problem? Please describe.**
We do not send our MSS (maximum segment size) to peer. https://en.wikipedia.org/wiki/Maximum_segment_size

The default value 536 is used if not set. Usually we could advertise larger value in order to send data is bigger chunks.

### PR DESCRIPTION
**Pull request by jukkar**
Fixes #30367

### CODE REVIEW

File: subsys/net/ip/tcp2.c [hunk 1]
```diff
@@ -1572,8 +1613,11 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
 	switch (conn->state) {
 	case TCP_LISTEN:
 		if (FL(&fl, ==, SYN)) {
+			/* Make sure our MSS is also send in the ACK */
```
Comment (pfalcon) [inline subsys/net/ip/tcp2.c:1616]: Typo in comment: "send" -> "sent".

File: subsys/net/ip/tcp2.c [hunk 2]
```diff
@@ -1572,8 +1613,11 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
...
+			conn->send_options.mss_found = true;
```
Comment (pfalcon) [inline subsys/net/ip/tcp2.c:1617]: Another concern is that reusing `struct tcp_options`, initially intended just for received options, also for send options, leads to a weird terminology, if using existing field names. IMHO, the fields should be renamed to more neutral first, e.g. `mss_found` -> `has_mss`.

(That's of course assuming the implementation approach taken in this PR is in the right direction.)

**Comment by galak** [comment]
@jukkar any updates here?  Is this still desired for 2.6?

**Comment by jukkar** [comment]
> any updates here? Is this still desired for 2.6?

This will not make to 2.6, requires more testing.

**Pull request closed without merging**